File: Solutions/0169/0169.swift
// Author: washing
// DateTime: 2021/1/26 18:48
// File: 0169.swift
// Desc: 


class Solution {
    func majorityElement(_ nums: [Int]) -> Int {
        var a = 0
        var c = 0
        for n in nums {
            if c == 0 {
                a = n
                c+=1
            } else if a == n {
                c+=1
            } else {
                c-=1
            }
        }
        return a
    }
}

File: Solutions/0169/0169.cs
// Author: washing
// DateTime: 2021/1/26 18:51
// File: 0169.cs
// Desc: 


public class Solution {
    public int MajorityElement(int[] nums) {
        int a=0;
        int c=0;
        int n;
        for (int i=0; i<nums.Length;i++){
            n = nums[i];
            if (c==0){
                a = n;
                c++;
            }else if (a==n){
                c++;
            }else{
                c--;
            }
        }
        return a;
    }
}